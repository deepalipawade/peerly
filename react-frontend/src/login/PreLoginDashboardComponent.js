import React from "react";
import { Col, Row, Container } from "core-components/grid/GridComponent";
import styled from "styled-components";
import PropTypes from "prop-types";

import PeerlyTextAndLogo from "shared-components/peerly-logo/PeerlyTextAndLogo";
import InformativeTextComponent from "login/InformativeTextComponent";
import LoginImageComponent from "login/LoginImageComponent";

const Wrapper = styled(Container)`
  height: 100vh;
  background-color: ${({ theme }) =>
    theme === "dark" ? "var(--atomic)" : "var(--white)"};
`;

const RowComponent = styled(Row)`
  height: calc(100% / 3);
`;

const PreLoginDashboardComponent = ({
  theme,
  className,
  informativeText,
  encouragementThought,
}) => (
  <Wrapper
    theme={theme}
    className={`d-sm-block d-xs-block d-md-none ${className}`}
    data-testid="officeInfo"
  >
    <RowComponent>
      <Col className="m-auto">
        <PeerlyTextAndLogo theme={theme} />
      </Col>
    </RowComponent>
    <RowComponent className="overflow-hidden">
      <Col className="p-0">
        <LoginImageComponent className="img-fluid" />
      </Col>
    </RowComponent>
    <RowComponent>
      <Col className="m-auto">
        <InformativeTextComponent
          theme={theme}
          informativeText={informativeText}
          encouragementThought={encouragementThought}
        />
      </Col>
    </RowComponent>
  </Wrapper>
);

PreLoginDashboardComponent.propTypes = {
  informativeText: PropTypes.string.isRequired,
  encouragementThought: PropTypes.string.isRequired,
  theme: PropTypes.oneOf(["dark", "light"]),
  className: PropTypes.string,
};

PreLoginDashboardComponent.defaultProps = {
  theme: "dark",
};

export default React.memo(PreLoginDashboardComponent);
