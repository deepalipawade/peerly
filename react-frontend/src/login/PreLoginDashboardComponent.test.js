import React from "react";
import { render } from "@testing-library/react";
import PreLoginDashboardComponent from "login/PreLoginDashboardComponent";

describe("pre login panel component test", () => {
  test("should render pre login panel component", () => {
    const { asFragment, getByTestId } = render(
      <PreLoginDashboardComponent
        informativeText="Lets Create the Office Positive"
        encouragementThought="thought"
      />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("officeInfo")).toHaveStyleRule(
      "background-color",
      "var(--atomic)"
    );
  });

  test("test case for style and classes", () => {
    const { getByTestId } = render(
      <PreLoginDashboardComponent
        theme="light"
        className="text-uppercase"
        informativeText="Lets Create the Office Positive"
        encouragementThought="thought"
      />
    );
    expect(getByTestId("officeInfo")).toHaveStyleRule(
      "background-color",
      "var(--white)"
    );
    expect(getByTestId("officeInfo")).toHaveClass("text-uppercase");
  });
});
