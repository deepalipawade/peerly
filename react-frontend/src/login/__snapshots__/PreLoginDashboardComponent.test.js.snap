// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pre login panel component test should render pre login panel component 1`] = `
<DocumentFragment>
  .c3 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  font-size: 48px;
  color: var(--white);
}

.c2 {
  text-align: center;
  margin-top: 15px;
  width: 83px;
  height: 83px;
  border: 3px solid var(--white);
  background-color: var(--black);
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c4 {
  color: var(--white);
  background-color: var(--black);
  font-size: 48px;
}

.c6 {
  width: 10%;
  background: var(--white);
}

.c5 {
  color: var(--white);
  background-color: var(--black);
}

.c0 {
  height: 100vh;
  background-color: var(--atomic);
}

.c1 {
  height: calc(100% / 3);
}

<div
    class="c0 d-sm-block d-xs-block d-md-none undefined container"
    data-testid="officeInfo"
    theme="dark"
  >
    <div
      class="c1 row"
    >
      <div
        class="m-auto col"
      >
        <div
          data-testid="peerlyLogoComponent"
        >
          <div
            class="c2 mx-auto"
            data-testid="Logo"
          >
            <div
              class="c3"
              data-testid="plusSignText"
              font-size="48px"
            >
              +
            </div>
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="c4"
            data-testid="PeerlyTextComponents"
            font-size="48px"
          >
            Peerly
          </div>
        </div>
      </div>
    </div>
    <div
      class="c1 overflow-hidden row"
    >
      <div
        class="p-0 col"
      >
        <img
          alt="login image"
          class="img-fluid img-fluid"
          src="/assets/images//cat-mobile-img.png"
        />
      </div>
    </div>
    <div
      class="c1 row"
    >
      <div
        class="m-auto col"
      >
        <div
          class="c5 text-center undefined"
          data-testid="info"
        >
          <h3>
            Lets Create the Office Positive
          </h3>
          <hr
            class="c6"
            data-testid="hrLine"
          />
          <h6>
            thought
          </h6>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
